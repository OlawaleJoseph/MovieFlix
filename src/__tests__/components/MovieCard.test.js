import { Button } from '@material-ui/core';
import Card from '../../components/MovieCard';
import shallowWrapper from '../../testSetup';

describe('Movie Card', () => {
  const movieInfo = {
    imgUrl: 'imageUrl',
  };

  const card = shallowWrapper(Card, movieInfo);

  test('should render', () => {
    expect(card).toBeTruthy();
    expect(card).toMatchSnapshot();
  });

  test('should have a image url', () => {
    const image = card.find('img');
    expect(image.prop('src')).toEqual(movieInfo.imgUrl);
  });

  test('should render Trailer button', () => {
    const buttons = card.find(Button);

    expect(buttons).toHaveLength(2);
    expect(buttons.first().text()).toContain('Trailer');
  });

  test('should render details button', () => {
    const buttons = card.find(Button);

    expect(buttons).toHaveLength(2);
    expect(buttons.last().text()).toContain('Details');
  });
});
