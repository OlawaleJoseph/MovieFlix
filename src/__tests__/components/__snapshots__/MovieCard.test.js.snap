// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Section should render Banner 1`] = `ShallowWrapper {}`;

exports[`Hero Section should render Card 1`] = `ShallowWrapper {}`;

exports[`Hero Section should render Movie Card 1`] = `ShallowWrapper {}`;

exports[`Movie Card should have a image url 1`] = `ShallowWrapper {}`;

exports[`Movie Card should render 1`] = `ShallowWrapper {}`;

exports[`card Section should render Movie Card 1`] = `ShallowWrapper {}`;

exports[`card Section should render Movie Card 2`] = `ShallowWrapper {}`;
